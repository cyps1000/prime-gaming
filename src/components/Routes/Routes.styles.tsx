/**
 *  Material UI Imports
 * @see https://material-ui.com/styles/basics/
 */
import { makeStyles } from "@material-ui/core/styles";

/**
 * Styles the component
 */
const useStyles = makeStyles((theme) => ({
  routes: {
    padding: 0,
    maxWidth: "none",
    height: "100%",
  },
}));

export { useStyles };
