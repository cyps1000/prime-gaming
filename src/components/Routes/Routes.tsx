/**
 * External Imports
 */
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";

/**
 * Component Imports
 */
import Main from "../Main";
import Navbar from "../Navbar";
import Body from "../Body";
import Footer from "../Footer";

/**
 *  Material UI Imports
 */
import Grid from "@material-ui/core/Grid";

/**
 * Imports the component styles
 */
import { useStyles } from "./Routes.styles";

/**
 * Displays the component
 */
const Routes: React.FC = () => {
  /**
   * Gets the component styles
   */
  const classes = useStyles();

  return (
    <Router>
      <Navbar />
      <Body>
        <Grid container className={classes.routes}>
          <Switch>
            <Route exact path="/" children={<Main text="Home" />} />
            <Route exact path="/news" children={<Main text="News" />} />
            <Route exact path="/about" children={<Main text="About" />} />
            <Route exact path="/contact" children={<Main text="Contact" />} />
          </Switch>
        </Grid>
      </Body>
      <Footer />
    </Router>
  );
};

export default Routes;
