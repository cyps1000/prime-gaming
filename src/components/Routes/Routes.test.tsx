/**
 * @see https://testing-library.com/docs/react-testing-library/intro
 */
import { render } from "@testing-library/react";

/**
 * Imports component
 */
import Routes from "./Routes";

/**
 * Default test
 */
it("has a Routes component", () => {});
