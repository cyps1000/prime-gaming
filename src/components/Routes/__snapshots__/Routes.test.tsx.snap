// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes Snapshot Test passes the snapshot test 1`] = `
"<div class=\\"makeStyles-root-111\\" data-testid=\\"navbar\\">
  <header class=\\"MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-nav-112 mui-fixed MuiPaper-elevation4\\">
    <div class=\\"MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters\\"><img class=\\"makeStyles-logo-119\\" src=\\"logo.png\\" alt=\\"logo\\">
      <div class=\\"makeStyles-search-120\\">
        <div class=\\"makeStyles-searchIcon-121\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\">
            <path d=\\"M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z\\"></path>
          </svg></div>
        <div class=\\"MuiInputBase-root makeStyles-inputRoot-122\\"><input placeholder=\\"Search...\\" type=\\"text\\" aria-label=\\"search\\" class=\\"MuiInputBase-input makeStyles-inputInput-123 makeStyles-expandAnimation-124\\" value=\\"\\"></div>
      </div>
      <div class=\\"makeStyles-actions-113\\">
        <div class=\\"makeStyles-container-125\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\">
            <path d=\\"M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69zM12 18c-3.31 0-6-2.69-6-6s2.69-6 6-6 6 2.69 6 6-2.69 6-6 6zm0-10c-2.21 0-4 1.79-4 4s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4z\\"></path>
          </svg><span class=\\"MuiSwitch-root makeStyles-switch-126\\"><span class=\\"MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-129 MuiSwitch-switchBase MuiSwitch-colorSecondary PrivateSwitchBase-checked-130 Mui-checked\\" aria-disabled=\\"false\\"><span class=\\"MuiIconButton-label\\"><input class=\\"PrivateSwitchBase-input-132 MuiSwitch-input\\" name=\\"checkedB\\" type=\\"checkbox\\" aria-label=\\"primary checkbox\\" value=\\"dark-theme\\" checked=\\"\\"><span class=\\"MuiSwitch-thumb\\"></span></span><span class=\\"MuiTouchRipple-root\\"></span></span><span class=\\"MuiSwitch-track\\"></span></span><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\">
            <path d=\\"M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69zM12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6c3.31 0 6 2.69 6 6s-2.69 6-6 6z\\"></path>
          </svg></div>
        <div data-testid=\\"language-switcher\\" class=\\"makeStyles-container-134\\"><img class=\\"makeStyles-flag-133\\" src=\\"https://www.countryflags.com/wp-content/uploads/romania-flag-png-xl.png\\" alt=\\"ro\\">
          <div><span class=\\"MuiSwitch-root\\"><span class=\\"MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-129 MuiSwitch-switchBase MuiSwitch-colorSecondary PrivateSwitchBase-checked-130 Mui-checked\\" aria-disabled=\\"false\\" data-testid=\\"language-switch-input\\"><span class=\\"MuiIconButton-label\\"><input class=\\"PrivateSwitchBase-input-132 MuiSwitch-input\\" type=\\"checkbox\\" value=\\"\\" checked=\\"\\"><span class=\\"MuiSwitch-thumb\\"></span></span><span class=\\"MuiTouchRipple-root\\"></span></span><span class=\\"MuiSwitch-track makeStyles-switchTrack-137\\"></span></span></div><img class=\\"makeStyles-flag-133\\" src=\\"https://www.countryflags.com/wp-content/uploads/united-kingdom-flag-png-large.png\\" alt=\\"uk\\">
        </div>
      </div>
      <div class=\\"makeStyles-menuItems-138\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\">Home</span><span class=\\"MuiTouchRipple-root\\"></span></button><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\">News</span><span class=\\"MuiTouchRipple-root\\"></span></button><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\">About</span><span class=\\"MuiTouchRipple-root\\"></span></button><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\">Contact</span><span class=\\"MuiTouchRipple-root\\"></span></button><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\">Sign In</span><span class=\\"MuiTouchRipple-root\\"></span></button><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\">Sign Up</span><span class=\\"MuiTouchRipple-root\\"></span></button></div>
    </div>
  </header>
</div>
<div class=\\"makeStyles-root-171\\">
  <div class=\\"MuiGrid-root makeStyles-routes-110 MuiGrid-container\\" data-testid=\\"router-grid\\">
    <div class=\\"makeStyles-root-172\\">
      <div class=\\"makeStyles-root-173 makeStyles-root-175\\"></div>
    </div>
  </div>
</div>
<header class=\\"MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-appBar-176 mui-fixed MuiPaper-elevation4\\" data-testid=\\"footer-default\\">
  <div class=\\"MuiToolbar-root MuiToolbar-regular makeStyles-toolbar-179 MuiToolbar-gutters\\">
    <div class=\\"MuiGrid-root MuiGrid-container\\">
      <div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3\\">
        <p class=\\"MuiTypography-root makeStyles-copyright-186 MuiTypography-body1\\">Â© 2021 Prime Gaming</p>
      </div>
      <div class=\\"MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-xs-center MuiGrid-grid-xs-6\\">
        <div class=\\"makeStyles-root-188\\"><button class=\\"MuiButtonBase-root MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root makeStyles-icon-189\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M5 3h14a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2m13 2h-2.5A3.5 3.5 0 0 0 12 8.5V11h-2v3h2v7h3v-7h3v-3h-3V9a1 1 0 0 1 1-1h2V5z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button><button class=\\"MuiButtonBase-root MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root makeStyles-icon-189\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M22.46 6c-.77.35-1.6.58-2.46.69.88-.53 1.56-1.37 1.88-2.38-.83.5-1.75.85-2.72 1.05C18.37 4.5 17.26 4 16 4c-2.35 0-4.27 1.92-4.27 4.29 0 .34.04.67.11.98C8.28 9.09 5.11 7.38 3 4.79c-.37.63-.58 1.37-.58 2.15 0 1.49.75 2.81 1.91 3.56-.71 0-1.37-.2-1.95-.5v.03c0 2.08 1.48 3.82 3.44 4.21a4.22 4.22 0 0 1-1.93.07 4.28 4.28 0 0 0 4 2.98 8.521 8.521 0 0 1-5.33 1.84c-.34 0-.68-.02-1.02-.06C3.44 20.29 5.7 21 8.12 21 16 21 20.33 14.46 20.33 8.79c0-.19 0-.37-.01-.56.84-.6 1.56-1.36 2.14-2.23z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button><button class=\\"MuiButtonBase-root MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root makeStyles-icon-189\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M7.8 2h8.4C19.4 2 22 4.6 22 7.8v8.4a5.8 5.8 0 0 1-5.8 5.8H7.8C4.6 22 2 19.4 2 16.2V7.8A5.8 5.8 0 0 1 7.8 2m-.2 2A3.6 3.6 0 0 0 4 7.6v8.8C4 18.39 5.61 20 7.6 20h8.8a3.6 3.6 0 0 0 3.6-3.6V7.6C20 5.61 18.39 4 16.4 4H7.6m9.65 1.5a1.25 1.25 0 0 1 1.25 1.25A1.25 1.25 0 0 1 17.25 8 1.25 1.25 0 0 1 16 6.75a1.25 1.25 0 0 1 1.25-1.25M12 7a5 5 0 0 1 5 5 5 5 0 0 1-5 5 5 5 0 0 1-5-5 5 5 0 0 1 5-5m0 2a3 3 0 0 0-3 3 3 3 0 0 0 3 3 3 3 0 0 0 3-3 3 3 0 0 0-3-3z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button><button class=\\"MuiButtonBase-root MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root makeStyles-icon-189\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M10 15l5.19-3L10 9v6m11.56-7.83c.13.47.22 1.1.28 1.9.07.8.1 1.49.1 2.09L22 12c0 2.19-.16 3.8-.44 4.83-.25.9-.83 1.48-1.73 1.73-.47.13-1.33.22-2.65.28-1.3.07-2.49.1-3.59.1L12 19c-4.19 0-6.8-.16-7.83-.44-.9-.25-1.48-.83-1.73-1.73-.13-.47-.22-1.1-.28-1.9-.07-.8-.1-1.49-.1-2.09L2 12c0-2.19.16-3.8.44-4.83.25-.9.83-1.48 1.73-1.73.47-.13 1.33-.22 2.65-.28 1.3-.07 2.49-.1 3.59-.1L12 5c4.19 0 6.8.16 7.83.44.9.25 1.48.83 1.73 1.73z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button></div>
      </div>
      <div class=\\"MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-xs-flex-end MuiGrid-grid-xs-3\\"><button class=\\"MuiButtonBase-root MuiIconButton-root makeStyles-iconButton-187\\" tabindex=\\"0\\" type=\\"button\\" data-testid=\\"minify-button\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button></div>
    </div><button class=\\"MuiButtonBase-root MuiFab-root makeStyles-fabButton-180 MuiFab-sizeSmall\\" tabindex=\\"0\\" type=\\"button\\" data-testid=\\"expand-list-button\\"><span class=\\"MuiFab-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M12 8l-6 6 1.41 1.41L12 10.83l4.59 4.58L18 14z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button>
    <div class=\\"MuiBox-root MuiBox-root-190 makeStyles-listWrapper-182\\">
      <nav class=\\"MuiList-root makeStyles-list-183 MuiList-padding\\">
        <div class=\\"MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button\\" tabindex=\\"0\\" role=\\"button\\" aria-disabled=\\"false\\">
          <div class=\\"MuiListItemText-root\\"><span class=\\"MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock\\">Terms of Service</span></div><span class=\\"MuiTouchRipple-root\\"></span>
        </div>
        <div class=\\"MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button\\" tabindex=\\"0\\" role=\\"button\\" aria-disabled=\\"false\\">
          <div class=\\"MuiListItemText-root\\"><span class=\\"MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock\\">Privacy policy</span></div><span class=\\"MuiTouchRipple-root\\"></span>
        </div>
        <div class=\\"MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button\\" tabindex=\\"0\\" role=\\"button\\" aria-disabled=\\"false\\">
          <div class=\\"MuiListItemText-root\\"><span class=\\"MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock\\">Cookie Policy</span></div><span class=\\"MuiTouchRipple-root\\"></span>
        </div>
        <div class=\\"MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button\\" tabindex=\\"0\\" role=\\"button\\" aria-disabled=\\"false\\">
          <div class=\\"MuiListItemText-root\\"><span class=\\"MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock\\">Contact us</span></div><span class=\\"MuiTouchRipple-root\\"></span>
        </div><button class=\\"MuiButtonBase-root MuiFab-root makeStyles-fabButton-180 MuiFab-sizeSmall\\" tabindex=\\"0\\" type=\\"button\\" data-testid=\\"shrink-list-button\\"><span class=\\"MuiFab-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button>
      </nav>
    </div>
  </div>
</header>"
`;
